Mathlib/Computability/Halting.lean

```diff
@@ -173,6 +173,18 @@ protected theorem not {p : α → Prop} (hp : ComputablePred p) : ComputablePred
         simp only [Bool.not_eq_true]
         cases f n <;> rfl⟩
 
+/-- The computable functions are closed under if-then-else definitions. -/
+theorem ite {f₁ f₂ : ℕ → ℕ} (hf₁ : Computable f₁) (hf₂ : Computable f₂)
+    {c : ℕ → Prop} [∀ k, Decidable (c k)] (hc : ComputablePred c) :
+    Computable fun k ↦ if c k then f₁ k else f₂ k :=
+  Computable.of_eq (by
+    show Computable fun k ↦ bif c k then f₁ k else f₂ k
+    exact Computable.cond (by
+      obtain ⟨h₀,h₁⟩ := hc
+      exact (Eq.to_iff <|congrArg _ (funext fun _ => decide_eq_decide.mpr <| Eq.to_iff rfl)).1 h₁)
+      hf₁ hf₂)
+    fun _ => by simp
```

Comment:
Here's a shorter proof:
```suggestion
    Computable fun k ↦ if c k then f₁ k else f₂ k := by
  simp_rw [← Bool.cond_decide]
  obtain ⟨inst, hc⟩ := hc
  convert Computable.cond hc hf₁ hf₂
```

Reply:
Nice! It looks like `convert` succeeded where I had expected `exact` to succeed (since there was no visible difference).